500-CNAB-dependencies.md

```diff
@@ -0,0 +1,100 @@
+
+## Depend on a named bundle
+
+The bundle depends on a specific named bundle that is known in advance.
+
+```json
+{
+    "name": "wordpress",
+    "dependencies": [
+        {
+            "requires":{
+                "bundle": "azure/mysql",
+                "version": {
+                    "range": "5.7.x",
+                    "prereleases": "true"
+                }
+            }
+        }
+    ]
+}
+```
+
+## Depend on a bundle that provides a capability
+
+This section is a placeholder and will be completed in a follow-up pull request.
+
+## Dependencies Metadata
+
+This specification introduces a `dependencies` array in the bundle.json
+that defines metadata necessary to specify a dependency.
+
+### requires
+
+The `requires` object defines the criteria for the dependent bundle:
+
+* `bundle`: A reference to a bundle in the format REGISTRY/NAME.
+* `version`: A set of criteria applied to the bundle version when selecting an
+    acceptable version of the bundle.
+    * `ranges`: An array of allowed version ranges. 
+    
+        Versions are specified using semver, with or without a leading "v". 
+        An `x` in the place of the minor or patch place can be used to specify 
+        a wildcard. Ranges can be specified by separating the two versions with 
+        a dash, the dash must be surrounded by spaces in order to disambiguate 
+        from prerelease tags.
+
+        Below are some example ranges:
+        * `1.2.3` - Requires an exact version.
+        * `1.2.x` - Restricts the major.minor to `1.2` and allows any patch.
+        * `1.x` - Restricts the major to `1` and allows any minor.
+        * `1.5.x - 3` - Allows a range of versions from `1.5` up to but not including 4.
+    * `prereleases`: A boolean field, defaults to `false`, which specifies if
+        prerelease versions of the bundle are allowed.
+
+# Q & A
+
+This isn't meant to be in the final spec but gives people a place to comment
+on anticipated questions.
+
+* Why didn't use you `~>` and other cute symbols?
+    * Because they have no global meaning across languages and frameworks and are hard to understand at a glance.
+* This doesn't say anything about how to use the dependency!
+    * This requires that the [outputs PR](https://github.com/deislabs/cnab-spec/pull/133) is merged. Once we have that, then given
+        a named bundle, we know exactly what the set of outputs of the bundle
+        will be and how to consume them.
+* Why didn't you fill out the "provides/requires" idea that you proposed a few weeks ago?
+    * This is a more straightforward idea for us to argue about first. 😅 We can
+        bikeshed over if it's a seperate spec, and on so many other wonderful topics
+        and on a very easy problem. Then if it's merged, then we can try tackling
+        the much more difficult problem of [Depend on a bundle that provides a capability](#depend-on-a-bundle-that-provides-a-capability).
```

Comment:
This is a very good approach. That whole virtual dependency thing was a nightmare in Debian, then a nightmare again in Gentoo. Not trying to figure it out in the rough drafts is probably good.